assets/js/common.js

```diff
@@ -264,6 +265,13 @@
 							},
 						};
 
+					if (options && options.icon) {
```

Comment:
It's still doesn't match the initial idea (e.g. it's not possible to override other settings via data-options, only the icon) but ok -- we can add a specific HTML attribute for this custom work, please change it to a specific attribute data-marker which passes the marker icon URL.

Reply:
Also please remember to test if map clustering (merging into a group when it's zoomed out) still works with a custom marker.